src/string_instruction_parser.yrl

```diff
@@ -1,43 +1,39 @@
 Header
 "%% Predicator Elixir".
 
-Terminals lit load comparator endcomparator jfalse jtrue '[' ']' ',' '&' string blank.
+Terminals lit load comparator endcomparator bang between 'or' '[' ']' ',' 'and'.
 
-Nonterminals predicates predicate value array array_elements.
+Nonterminals predicates predicate_group predicate variable array array_elements array_value.
 
 Rootsymbol predicates.
 
-predicates -> predicate : '$1'.
-predicates -> predicate jfalse predicate : ['$1', jfalse, '$3']. %% jfalse
-predicates -> predicates jfalse predicate : {'$1', jfalse, '$3'}.
-predicates -> predicate jtrue predicate : ['$1', jtrue, '$3']. %% jtrue
-predicates -> predicates jtrue predicate : {'$1', jtrue, '$3'}.
-
-predicate -> load endcomparator : [unwrap('$1'), unwrap('$2')].
-predicate -> lit endcomparator : [unwrap('$1'), unwrap('$2')].
-predicate -> string endcomparator : [unwrap_string('$1'), unwrap('$2')].
-predicate -> lit comparator load : [unwrap('$1'), unwrap('$3'), unwrap('$2')].
-predicate -> load comparator lit : [unwrap('$1'), unwrap('$3'), unwrap('$2')].
-predicate -> lit comparator lit : [unwrap('$1'), unwrap('$3'), unwrap('$2')].
-predicate -> load comparator load : [unwrap('$1'), unwrap('$3'), unwrap('$2')].
-predicate -> load comparator lit '&' lit : [unwrap('$1'), unwrap('$3'), unwrap('$5'), unwrap('$2')].
-predicate -> lit comparator lit '&' lit : [unwrap('$1'), unwrap('$3'), unwrap('$5'), unwrap('$2')].
-predicate -> load comparator array : [unwrap('$1'), [<<"array">>, '$3'], unwrap('$2')].
-predicate -> lit comparator array : [unwrap('$1'), [<<"array">>, '$3'], unwrap('$2')].
-predicate -> string comparator array : [unwrap_string('$1'), [<<"array">>, '$3'], unwrap('$2')].
-predicate -> string comparator string : [unwrap_string('$1'), unwrap_string('$3'), unwrap('$2')].
-predicate -> load comparator string : [unwrap('$1'), unwrap_string('$3'), unwrap('$2')].
+predicates -> predicate_group : '$1'.
+predicates -> bang predicate_group : lists:append('$2', [[<<"not">>]]).
+
+predicate_group -> predicate : '$1'.
+predicate_group -> predicate 'and' predicates : lists:append('$1', [jump(jfalse, '$3') | '$3']).
+predicate_group -> predicate 'or' predicates : lists:append('$1', [jump(jtrue, '$3') | '$3']).
+
+predicate -> lit : [unwrap('$1')].
+predicate -> load : [unwrap('$1'), [<<"to_bool">>]].
+predicate -> variable endcomparator : ['$1', unwrap('$2')].
+predicate -> variable comparator variable : ['$1', '$3', unwrap('$2')].
+predicate -> variable between lit 'and' lit : ['$1', unwrap('$3'), unwrap('$5'), unwrap('$2')].
+predicate -> variable comparator array : ['$1', [<<"array">>, '$3'], unwrap('$2')].
+
+variable -> lit : unwrap('$1').
+variable -> load : unwrap('$1').
```

Comment:
cleaned this up a little, if I didn't I would have had to add bang to every possibility and the amount of matches doubled 😱